src/java/org/apache/cassandra/hints/HintVerbHandler.java

```diff
@@ -101,7 +102,16 @@ else if (!StorageProxy.instance.appliesLocally(hint.mutation))
         else
         {
             // the common path - the node is both the destination and a valid replica for the hint.
-            hint.applyFuture().addCallback(o -> respond(message), e -> logger.debug("Failed to apply hint", e));
+            hint.applyFuture().addCallback(
+                    o -> {
+                        HintsServiceMetrics.hintsApplySucceeded.mark();
```

Comment:
@maoling can we do this _after_ we do `respond(message)`? What if `respond(message)` fails?